xmodule/modulestore/xml.py

```diff
@@ -239,6 +239,24 @@ def load_item(usage_key, for_parent=None):
             **kwargs
         )
 
+    # pylint: disable=keyword-arg-before-vararg
+    def construct_xblock_from_class(self, cls, scope_ids, field_data=None, *args, **kwargs):
+        """
+        Construct a new xblock of type cls, mixing in the mixins
+        defined for this application.
+        """
+        if field_data:
+            # Currently, *some* XBlocks (those with XmlMixin) use XmlMixin.parse_xml() which instantiates
+            # its own key value store for field data. That is something which should be left to the runtime, for
+            # consistent behavior across all XBlocks, not controlled by individual XBlock implementations.
+            # We cannot just ignore field_data here though, because parse_xml may have pre-loaded data into it that we
+            # would otherwise lose.
```

Comment:
This is a great warning, but now I'm curious to understand–why does `XmlMixin.parse_xml()` create its own `KvsFieldData`?

Reply:
I honestly don't know. Possibly because at the time there was no API for asking the runtime to create a new block without passing in a field data instance? (But now there is...)

And yes, I hope this warning will help others; I'm adding it because it was difficult to debug some inconsistent behavior I was seeing until I finally figured out that during import or parsing OLX, some XBlocks were using totally different FieldData than what the runtime was providing for them to use.